imod/typing/grid.py

```diff
@@ -398,3 +398,27 @@ def decorator(*args, **kwargs):
         return x
 
     return decorator
+
+
+def is_planar_grid(
+    grid: xr.DataArray | xr.Dataset | xu.UgridDataArray | xu.UgridDataset,
+) -> bool:
+    # Returns True if the grid is planar. It has then a layer coordinate with
+    # length 1 and value 0, or an empty layer coordinate axis, or no layer coordinate at all
+    if "layer" not in grid.coords:
+        return True
+    if grid["layer"].shape == ():
+        return True
+    if grid["layer"][0] == 0 and len(grid["layer"]) == 1:
+        return True
+    return False
```

Comment:
Nice, I think a requirement for this one also that it is spatial_2D. Settings stored in xr.DataArrays/xr.Datasets without a dimension now also return True. 
For example:
```python
ds = xr.Dataset()
ds["save_flows"] = True

print(is_planar_grid(ds))
# Prints True, but should print False

```

So I suggest something like:

```suggestion
def is_planar_grid(
    grid: xr.DataArray | xr.Dataset | xu.UgridDataArray | xu.UgridDataset,
) -> bool:
    if not is_spatial_2D(grid):
         return False
    # Returns True if the grid is planar. It has then a layer coordinate with
    # length 1 and value 0, or an empty layer coordinate axis, or no layer coordinate at all
    if "layer" not in grid.coords:
        return True
    if grid["layer"].shape == ():
        return True
    if grid["layer"][0] == 0 and len(grid["layer"]) == 1:
        return True
    return False
```

Reply:
done, nice catch.